frontend/readhistory.lua

```diff
@@ -199,6 +200,21 @@ function ReadHistory:updateItemByPath(old_path, new_path)
     end
 end
 
+--- Updates the history list after renaming/moving a folder.
+function ReadHistory:updateItemsByPath(old_path, new_path)
+    local history_updated
+    for i, v in ipairs(self.hist) do
+        local file, count = v.file:gsub(old_path, new_path)
+        if count == 1 then
+            self.hist[i].file = file
```

Comment:
About the :gsub(), I guess you should try to match at start, and limit the nb of replacements to 1 (4th parameter, or no need if matching at start).
So sick cases like renaming bits of `/mnt/onboard/Library/Fiction/mnt/onboard/Library/myfile.epub` are supported :)